### ISSUE TITLE
Xamarin support

### ISSUE THREAD

**Issue by clairernovotny** [microsoft/vstest#1706]
The VS Test platform should support Xamarin platforms as first-class citizens for discovering and executing tests. It should work similarly to the way the UWP-based ones work, where it can deploy to the device for discovery and execution.

**Comment by dotMorten** [comment]
I'd also like identical or near-identical commandline options for deploying, executing and retrieving a TRX report like vstest.console.exe (obviously with specific parameters to deploy to android or ios devices/simulators)

**Comment by dotMorten** [comment]
I've been hacking around in testfx to build a TestAdapter that works on Android. The good news is it was only VERY few changes that needed to be made, to allow you to use it. Really if you could release a netstd2.0 compatible TestAdapter library, that would be the first huge step forward, since all you would t hen need to add yourself is a simple little test runner on top of it, considering the test execution etc is already done by the adapter.

**Comment by pvlakshm** [comment]
Xamarin support can be enabled in a manner similar to what is used for UWP today.
It means that the `vstest` platform be able to deploy-to-target and run the tests there, where the target could be a emulator/simular/device running Android, or iOS via a paired Mac.
The vstest platform can support this via host extensibilty. See here: [https://github.com/Microsoft/vstest-docs/blob/master/RFCs/0001-Test-Platform-Architecture.md](). However we have had only UWP leveraging this model; so there might still be some wrinkles/issues - but those can be addressed.
In summary, this will require a host provider. Contributions welcome!

**Comment by onovotny** [comment]
@pranavkm 
> Contributions welcome!

I would say this has to be prioritized and done by the VSTest team itself, working with the Xamarin team as appropriate. VSTest is missing major platforms, and ones that are first-party at that.

**Comment by dotMorten** [comment]
Any update on making this happen soon? I'd think this should be a min-ship feature for .NET 5

**Comment by nohwnd** [comment]
@pavelhorak for timeline and prioritization.

**Comment by pavelhorak** [comment]
Let me take this up with PM.

**Comment by pavelhorak** [comment]
We might consider this for .NET 5, but now we're focusing on higher priority issues.

**Comment by nohwnd** [comment]
This is a new feature and won't be implemented, we are focusing on adding new features to Testing.Platform instead. https://aka.ms/testingplatform
